core/src/main/java/feast/core/service/JobCoordinatorService.java

```diff
@@ -108,6 +112,11 @@ public void Poll() throws InvalidProtocolBufferException {
           .collect(Collectors.groupingBy(FeatureSet::getSource))
           .forEach(
               (source, setsForSource) -> {
+                // Sources with same type and config in different Feature Sets are different
+                // objects.
+                // Make sure that we are dealing with the same source object when spawning jobs.
```

Comment:
This would not get the same object, but it would get objects with the same type and configuration. As long as we are clear that we wont use the object Id anywhere, then that should be fine.